js/packages/mobile-wallet-adapter-walletlib/android/src/main/java/com/solanamobile/mobilewalletadapterwalletlib/reactnative/SolanaMobileWalletAdapterWalletLibModule.kt

```diff
@@ -83,66 +99,33 @@ class SolanaMobileWalletAdapterWalletLibModule(val reactContext: ReactApplicatio
         }
     }
 
-    fun convertReadableMapToConfig(config: ReadableMap): MobileWalletAdapterConfig {
-        val supportsSignAndSendTransactions = config.getBoolean("supportsSignAndSendTransactions")
-        val maxTransactionsPerSigningRequest = config.getInt("maxTransactionsPerSigningRequest")
-        val maxMessagesPerSigningRequest = config.getInt("maxMessagesPerSigningRequest")
-        val readableArray = config.getArray("supportedTransactionVersions")
-        val supportedTransactionVersions = mutableListOf<Any>()
-        for (i in 0 until readableArray?.size()!!) {
-            when (readableArray?.getType(i)) {
-                ReadableType.Number -> supportedTransactionVersions.add(readableArray.getInt(i))
-                ReadableType.String -> supportedTransactionVersions.add(readableArray.getString(i))
-                else -> throw IllegalArgumentException("Unsupported type in supportedTransactionVersions array")
-            }
-        }
-
-        val noConnectionWarningTimeoutMs = config.getDouble("noConnectionWarningTimeoutMs").toLong()
-
-        return MobileWalletAdapterConfig(
-            supportsSignAndSendTransactions,
-            maxTransactionsPerSigningRequest,
-            maxMessagesPerSigningRequest,
-            supportedTransactionVersions.toTypedArray(),
-            noConnectionWarningTimeoutMs
-        )
-    }
-
-    // Converts a React ReadableArray into a Kotlin ByteArray.
-    // Expects ReadableArray to be an Array of ints, where each int represents a byte.
-    private fun convertFromReactByteArray(reactByteArray: ReadableArray): ByteArray {
-        return ByteArray(reactByteArray.size()) { index ->
-            reactByteArray.getInt(index).toByte()
-        }
-    }
-
-    // Converts a Kotlin ByteArray into a React ReadableArray of ints.
-    private fun convertToReactByteArray(byteArray: ByteArray): ReadableArray {
-        return Arguments.createArray().apply {
-            byteArray.forEach {
-                this.pushInt(it.toInt())
-            }
-        }
-    }
-
     @ReactMethod
     fun createScenario(
         walletName: String,
         uriStr: String,
         config: ReadableMap
-    ) {
+    ) = launch {
         val uri = Uri.parse(uriStr)
 
+        // TODO: this is dirty, need some stateful object/data to know what state we are in.
```

Comment:
Definitely a TODO for now. Maybe this is just something we can follow up once we get feedback from first wallets that implement.